pkg/system/phase2_creating.go

```diff
@@ -152,6 +159,46 @@ func (r *Reconciler) ReconcilePhaseCreatingForMainClusters() error {
 	return nil
 }
 
+func (r *Reconciler) UpgradeDb() error {
```

Comment:
maybe rename to `UpgradeDbPostgres16`, since it is very specific for that version

Reply:
This in general should handle any future version of pg upgrade according to `options.DBImage `. The db image upgrade is decided based the current image and the image provided in the options.go.

Reply:
renaming it to `UpgradeDbPostgres16`